File: tests/fixtures/sample.scala
import scala.collection.mutable.ListBuffer

case class Config(baseUrl: String, timeout: Int)

trait Loggable
abstract class BaseClient

class HttpClient(config: Config) extends BaseClient with Loggable {
  val source: Config = config
  var fallback: BaseClient = null

  def get(path: String): String = {
    buildRequest("GET", path)
  }

  def post(path: String, body: String): String = {
    buildRequest("POST", path)
  }

  private def buildRequest(method: String, path: String): String = {
    s"$method ${config.baseUrl}$path"
  }
}

object HttpClientFactory {
  def create(baseUrl: String): HttpClient = {
    new HttpClient(Config(baseUrl, 30))
  }
}

File: tests/fixtures/sample.kt
import kotlinx.coroutines.delay
import kotlin.math.max

data class Config(val baseUrl: String, val timeout: Int)

class HttpClient(private val config: Config) {
    fun get(path: String): String {
        return buildRequest("GET", path)
    }

    fun post(path: String, body: String): String {
        return buildRequest("POST", path)
    }

    private fun buildRequest(method: String, path: String): String {
        return "$method ${config.baseUrl}$path"
    }
}

interface Loggable {
    fun log()
}

open class BaseProcessor

class Result<T>

class DataProcessor : BaseProcessor(), Loggable {
    var current: Result<DataProcessor> = Result()

    fun run(input: DataProcessor): Result<DataProcessor> {
        return current
    }

    override fun log() {}
}

class LoggingList<T>(inner: MutableList<T>) : MutableList<T> by inner

fun createClient(baseUrl: String): HttpClient {
    val config = Config(baseUrl, 30)
    return HttpClient(config)
}

enum class ChatType {
    NORMAL,
    GROUP,
    SYSTEM
}
